source/agora/consensus/data/Block.d

```diff
@@ -538,10 +519,12 @@ version (unittest)
         Transactions txs, Enrollment[] enrollments = null,
         Hash random_seed = Hash.init, uint[] missing_validators = null,
         KeyPair[] keys = genesis_validator_keys,
-        ulong delegate (PublicKey) cycleForValidator = (PublicKey k) => defaultCycleZero(k)) @safe nothrow
+        ulong delegate (PublicKey) cycleForValidator = (PublicKey k) => defaultCycleZero(k),
+        ulong timestamp = ulong.max) @safe nothrow
     {
-        auto block = makeNewBlock(prev_block, txs, enrollments, random_seed,
-            missing_validators);
+        auto block = makeNewBlock(prev_block, txs,
+                (timestamp == ulong.max) ? prev_block.header.timestamp + 1 : timestamp,
```

Comment:
Shouldn't this be `+ ConsensusParams.BlockInterval`?

Reply:
1. a lot of unit tests will not adjust the clock and lot of them are not related to 'time consensus', and by adding only 1 will ensure that those unrelated unit tests will pass

2. the reason why I didn't add  ConsensusParams.BlockInterval is because at this level most unit tests have no idea about what the ConsensusParams are, and they are just not using it.

3. if some unit tests would like to change the clock or the block creation time, it can pass any timestamp it wants

Reply:
Could you add a comment summarizing this to the code then ?
E.g. "// This should really be `+ ConsensusParams.BlockInterval` instead of `+ 1`, but ..."